lib/taskana-core/src/test/java/acceptance/task/ServiceLevelPriorityAccTest.java

```diff
@@ -256,4 +260,163 @@ void testSetPlannedPropertyOnEmptyTasksList()
+    Task finalTask = task;
+    assertThatThrownBy(
+        () -> {
+            taskService.updateTask(finalTask);
+        })
+        .isInstanceOf(InvalidArgumentException.class);
+
+    // update due and planned as expected.
+    task = taskService.getTask(taskId);
+    task.setDue(planned.plus(Duration.ofDays(3)));
+    DaysToWorkingDaysConverter converter = DaysToWorkingDaysConverter.initialize();
+    long days = converter.convertWorkingDaysToDays(task.getDue(), -1);
+    task.setPlanned(task.getDue().plus(Duration.ofDays(-1)));
+    task = taskService.updateTask(task);
+    days = converter.convertWorkingDaysToDays(task.getDue(), -1);
+    assertThat(task.getPlanned()).isEqualTo(task.getDue().plus(Duration.ofDays(days)));
+
+    // update due and planned as expected.
+    task = taskService.getTask(taskId);
+    task.setDue(planned.plus(Duration.ofDays(3)));
+    task.setPlanned(null);
+    task = taskService.updateTask(task);
+    days = converter.convertWorkingDaysToDays(task.getDue(), -1);
+    assertThat(task.getPlanned()).isEqualTo(task.getDue().plus(Duration.ofDays(days)));
+  }
+
+  @WithAccessId(
+      userName = "admin",
+      groupNames = {"group_2"})
+  @Test
+  void testSetPlannedPropertyOnSingleTaskUpdateWithNulls()
+      throws NotAuthorizedException, TaskNotFoundException, InvalidArgumentException,
+          ConcurrencyException, InvalidStateException, ClassificationNotFoundException,
+          AttachmentPersistenceException {
+    String taskId = "TKI:000000000000000000000000000000000002";
+    final Instant planned = getInstant("2020-05-03T07:00:00");
+    Task task = taskService.getTask(taskId);
+    
+    task.setPlanned(null);
+    task = taskService.updateTask(task);
+    DaysToWorkingDaysConverter converter = DaysToWorkingDaysConverter.initialize();
+    long days = converter.convertWorkingDaysToDays(task.getPlanned(), 1);
+    assertThat(task.getDue()).isEqualTo(task.getPlanned().plus(Duration.ofDays(days)));
+
+    task.setDue(null);
+    task = taskService.updateTask(task);
+    days = converter.convertWorkingDaysToDays(task.getPlanned(), 1);
+    assertThat(task.getDue()).isEqualTo(task.getPlanned().plus(Duration.ofDays(days)));
+
+    task.setPlanned(planned.plus(Duration.ofDays(13)));
+    task.setDue(null);
+    task = taskService.updateTask(task);
+    days = converter.convertWorkingDaysToDays(task.getPlanned(), 1);
+    assertThat(task.getDue()).isEqualTo(task.getPlanned().plus(Duration.ofDays(days)));
+
+    task.setDue(planned.plus(Duration.ofDays(13)));
+    task.setPlanned(null);
+    task = taskService.updateTask(task);
+    days = converter.convertWorkingDaysToDays(task.getDue(), -1);
+    assertThat(task.getDue()).isEqualTo(planned.plus(Duration.ofDays(13)));
+    assertThat(task.getPlanned()).isEqualTo(task.getDue().plus(Duration.ofDays(days)));
+  }
+
+  @WithAccessId(
+      userName = "admin",
+      groupNames = {"group_2"})
+  @Test
+  void testSetPlannedPropertyOnAllTasks()
```

Comment:
This test has no side effects because I added a @beforeEach method that resets the database before each testcase.
However, this could have impact on performance of our build process. Therefore, I think I should remove the @beforeEach method and move this specific testcase into a class of its own.